bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -113,14 +113,27 @@ public synchronized void registered(SelectionKey key)
   @Override
   public void unregistered(SelectionKey key)
   {
-    serverHelperExecutor.shutdown();
-    storageHelperExecutor.shutdown();
-    try {
-      serverHelperExecutor.awaitTermination(5000, TimeUnit.MILLISECONDS);
-    } catch (InterruptedException ex) {
-      logger.debug("Executor Termination", ex);
+    for (LogicalNode ln : subscriberGroups.values()) {
+      ln.boot(eventloop);
```

Comment:
It helps to avoid RejectedExecutionException as it adds Subscriber.unregister() to the event loop tasks queue. Subscriber.unregister() uses serverHelperExecutor, so it will not be shutdown until all Subscribers are unregistered.